editor/project_settings_editor.cpp

```diff
@@ -287,6 +287,8 @@ void ProjectSettingsEditor::_add_feature_overrides() {
 	presets.insert("s3tc");
 	presets.insert("etc2");
 	presets.insert("editor");
+	presets.insert("editor_hint");
+	presets.insert("editor_runtime");
```

Comment:
I wonder if we shouldn't have just `editor` and `editor_runtime` (so keep using `editor` for what is now `editor_hint` in this PR).

This would slightly break compat on the behavior of `editor`, but IMO the current behavior is confusing, and the two new feature tags make it worse as it's not obvious to users what's the difference between `editor` and `editor_hint`/`editor_runtime`.

Reply:
`editor` is useful for any code that's supposed to run only in editor build, so that the code does not accidentally slip to exported project, thus making it closer to `editor_runtime`. From my experience at least, idk how other people use it.

Reply:
> I guess one could use debug for that?

Unless they ship game using debug build (which can happen sometimes ;)